src/main/scala/org/neo4j/spark/Neo4jConfig.scala

```diff
@@ -1,34 +1,44 @@
 package org.neo4j.spark
 
 import org.apache.spark.SparkConf
-import org.neo4j.driver.v1.{Driver, AuthTokens, Config, GraphDatabase}
+import org.neo4j.driver.{AccessMode, AuthToken, AuthTokens, Config, Driver, GraphDatabase, Session, SessionConfig}
 
 /**
  * @author mh
  * @since 02.03.16
  */
-case class Neo4jConfig(val url: String, val user: String = "neo4j", val password: Option[String] = None, encryption:Boolean) {
+case class Neo4jConfig(val url: String,
+                       val user: String = "neo4j",
+                       val password: Option[String] = None,
+                       val database: Option[String] = None,
+                       val encryption: Boolean) {
 
-  def boltConfig(): Config = if (encryption) Config.build().withEncryption().toConfig else Config.build().withoutEncryption().toConfig
+  private def boltConfig(): Config = if (encryption) Config.builder().withEncryption().build() else Config.builder().withoutEncryption().build()
 
   def driver(config: Neo4jConfig) : Driver = config.password match {
-    case Some(pwd) => GraphDatabase.driver(config.url, AuthTokens.basic(config.user, pwd), boltConfig())
-    case _ => GraphDatabase.driver(config.url, boltConfig())
+    case Some(pwd) => driver(config.url, AuthTokens.basic(config.user, pwd))
+    case _ => driver(config.url, AuthTokens.none())
   }
 
-  def driver() : Driver = driver(this)
+  def driver(): Driver = driver(this)
 
-  def driver(url: String): Driver = GraphDatabase.driver(url, boltConfig())
+  def driver(url: String, authToken: AuthToken): Driver = GraphDatabase.driver(url, authToken, boltConfig())
+
+  def sessionConfig(write: Boolean = false): SessionConfig = database.map { SessionConfig.builder().withDatabase(_) }
+    .getOrElse(SessionConfig.builder())
+    .withDefaultAccessMode(if (write) AccessMode.WRITE else AccessMode.READ)
+    .build()
 
 }
 
 object Neo4jConfig {
-  val prefix = "spark.neo4j.bolt."
```

Comment:
should we as a fallback support the old syntax for a version or two?

Reply:
Oki